S/Sundials/build_tarballs.jl

```diff
@@ -1,31 +1,32 @@
 using BinaryBuilder
 
 name = "Sundials"
-version = v"3.1.1"
+version = v"5.1.0"
 
 # Collection of sources required to build SundialsBuilder
 sources = [
     "https://github.com/LLNL/sundials/archive/v$(version).tar.gz" =>
-    "d03fca24d7adbfe52e05bd4e61340251e1cf56927540cc77a1ca817716091166",
+    "101be83221f9a0ab185ecce04d003ba38660cc71eb81b8a7cf96d1cc08b3d7f9",
     "./bundled",
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/sundials-*/
-patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
+# patch -p0 < $WORKSPACE/srcdir/patches/Sundials_windows.patch
 
 CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}")
 CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release -DEXAMPLES_ENABLE_C=OFF)
 CMAKE_FLAGS+=(-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR="$prefix/include" -DKLU_LIBRARY_DIR="$libdir")
-CMAKE_FLAGS+=(-DBLAS_ENABLE=OFF -DLAPACK_ENABLE=OFF)
+CMAKE_FLAGS+=(-DLAPACK_ENABLE=ON)
 
-#if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
-#    patch -p0 < $WORKSPACE/srcdir/patches/Sundials_ilp64.patch
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas64_.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
-#else
-#    CMAKE_FLAGS+=(-DBLAS_LIBRARIES="${libdir}/libopenblas.${dlext}" -DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
-#fi
+if [[ ${nbits} == 64 ]] && [[ ${target} != aarch64* ]]; then
+    atomic_patch -p1 $WORKSPACE/srcdir/patches/Sundials_Fortran.patch
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas64_.${dlext}")
+else
+    CMAKE_FLAGS+=(-DLAPACK_LIBRARIES="${libdir}/libopenblas.${dlext}")
+fi
+export CFLAGS="-lgfortran -lquadmath"
```

Comment:
If I'm not mistaken, it looks like `libquadmath` is available only for x86_64 and i686 platforms, so this should become:
```suggestion
export CFLAGS="-lgfortran"
if [[ "${target}" == i686-* ]] || [[ "${target}" == x86_64-* ]]; then
    export CFLAGS="${CFLAGS} -lquadmath"
fi
```

Reply:
Does gfortran have some sort of a flag that tells you which libraries to link to? I wish it did.